app/src/androidTest/java/com/lightricks/feedexercise/data/FeedRepositoryTest.kt

```diff
@@ -1,13 +1,202 @@
+import com.lightricks.feedexercise.network.TemplatesMetadata
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import io.reactivex.Single
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Response
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
 import org.junit.runner.RunWith
+import retrofit2.HttpException
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.create
+import retrofit2.http.GET
+import java.io.IOException
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
 @RunWith(AndroidJUnit4::class)
 class FeedRepositoryTest {
-   //todo: add the tests here
+
+
+
+    private lateinit var userDao: FeedItemDao
+    private lateinit var db: FeedDatabase
+    private lateinit var appContext: Context
+    private lateinit var jsonContent: String
+    private lateinit var serviceResponse: TemplatesMetadata
+
+    @Before
+    fun createDb() {
+        appContext = ApplicationProvider.getApplicationContext<Context>()
+        db = Room.inMemoryDatabaseBuilder(
+            appContext, FeedDatabase::class.java).build()
+        userDao = db.feedItemDao()
+    }
+
+    @Before
+    fun getJsonStr(){
+        jsonContent = appContext.assets
+            .open("get_feed_response.json")
+            .bufferedReader()
+            .use{ it.readText() }
+        val moshi = Moshi.Builder()
+            .add(KotlinJsonAdapterFactory())
+            .build()
+        val adapter = moshi.adapter<TemplatesMetadata>(TemplatesMetadata::class.java)
+        serviceResponse = adapter.fromJson(jsonContent)!!
+    }
+
+    @After
+    @Throws(IOException::class)
+    fun closeDb() {
+        db.close()
+    }
+
+    @get:Rule
+    var instantExecutorRule = InstantTaskExecutorRule()
+
+
+    class MockInterceptor : Interceptor {
+
+        override fun intercept(chain: Interceptor.Chain): Response {
+                return chain.proceed(chain.request())
+                    .newBuilder()
+                    .code(404)
+                    .protocol(Protocol.HTTP_2)
+                    .build()
+        }
+    }
+
+    interface MockFeedApiService : FeedApiService {
+        @GET("feed.json")
+        override fun getFeed() : Single<TemplatesMetadata>
+
+        object FeedApi {
+            val service = Retrofit.Builder()
+                .addConverterFactory(MoshiConverterFactory.create())
+                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+                .client(OkHttpClient.Builder()
+                    .addInterceptor(MockInterceptor()).build())
+                .baseUrl("https://google.com")
+                .build()
+                .create<MockFeedApiService>()
+        }
+    }
+    @Test
+    fun refreshShouldFailWith404Code(){
```

Comment:
This reads like we expect failure by default. What do you think about `refreshFailsUponHttpException`?